### PR TITLE
Added 787 in Java

### PR DESCRIPTION
**Pull request by spattk**

### CODE REVIEW

File: java/787_Cheapest_Flight_Within_K_Stops.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        temp[src] = 0;
+        
+        for(int i=0; i<=k; i++){
+            computePrice(flights, prices, temp);
+            copyTempToPrice(prices, temp);
+        }
+        
+        return prices[dst] == Integer.MAX_VALUE ? -1 : prices[dst];
+    }
+}
```
Comment (qiyuangong) [inline java/787_Cheapest_Flight_Within_K_Stops.java:41]: Empty line at the end.

File: java/787_Cheapest_Flight_Within_K_Stops.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            if(prices[u] != Integer.MAX_VALUE){
+                if(prices[u] + price < temp[v]){
+                    temp[v] = prices[u] + price;
+                }
+            }
+        }
+    }
+    
+    public void copyTempToPrice(int[] prices, int[] temp){
+        for(int i=0; i<prices.length; i++){
```
Comment (qiyuangong) [inline java/787_Cheapest_Flight_Within_K_Stops.java:19]: Code style.

File: java/787_Cheapest_Flight_Within_K_Stops.java (hunk 1 above)
Comment (spattk) [reply java/787_Cheapest_Flight_Within_K_Stops.java:41]: Sorry, do you mean an empty line should be added as well?

File: java/787_Cheapest_Flight_Within_K_Stops.java (hunk 2 above)
Comment (spattk) [reply java/787_Cheapest_Flight_Within_K_Stops.java:19]: Sorry, I didn't get this one too.
Do you mean to convert this to a for each loop or something else?

Comment (qiyuangong) [reply java/787_Cheapest_Flight_Within_K_Stops.java:19]: Oh. My fault! This code style is also good.

File: java/787_Cheapest_Flight_Within_K_Stops.java (hunk 1 above)
Comment (qiyuangong) [reply java/787_Cheapest_Flight_Within_K_Stops.java:41]: Yes. Otherwise, github will give you a mark in that file.

**Pull request merged**